### ISSUE TITLE
No Ubuntu 22 (Jammy) support

### ISSUE THREAD

**Issue by JojoBr0** [crowdsecurity/cs-nginx-bouncer#38]
## Describe the bug

There is no available package for ubuntu 22 LTS from the official [repositories](https://doc.crowdsec.net/docs/getting_started/install_crowdsec#install-our-repositories).

Also if you want to install it manually, there is no candidate for the required libnginx-mod-http-lua package.

**Comment by LaurenceJJones** [comment]
Since ubuntu has dropped support for `libnginx-mod-http-lua` package we cannot compile for this. From the ubuntu changelogs
```
 Remove the Lua modules from NGINX (Server Team Decision) - future support
      for Lua module now requires resty-core from OpenResty, meaning that if
      we want to continue to support the Lua module, we have to start becoming
      OpenResty - users should just use OpenResty at this point for Lua.
      Changes made for this removal:
       + d/control:
         + Remove lua module from dependencies, and binary build item.
         + Add "Breaks" line for nginx-lua for older versions of NGINX.
           This is added to the nginx metapackage and nginx-extras.
       + d/rules: Remove Lua module from the build flags for -extras.
       + d/http-lua, d/modules/{,patches/,watch/}nginx-lua: Remove Lua
         modules, watch file, module patches.
       + d/modules/control: Remove Lua module from definitions
       + d/copyright: Remove lua module.
       + d/tests/{control,lua/}: Remove Lua test entirely, remove
         dependencies on any test which request
         libnginx-mod-http-lua as it's gone.
```
They are just advising you to install openresty moving forward

**Comment by LaurenceJJones** [comment]
Closing due to out of our hands, but will pin for visibility.

**Comment by luizvaz** [comment]
Sorry to re-open this closed issue.

The solution to this issue is _ppa/odrej_ repositories.

The steps below work on Ubuntu 22.04:
```
sudo apt install crowdsec-nginx-bouncer
sudo apt install nginx lua5.1 libnginx-mod-http-lua luarocks gettext-base lua-cjson
sudo apt install crowdsec-nginx-bouncer
```

Maybe you can update the page https://docs.crowdsec.net/u/bouncers/nginx

Best regards,
Luiz Vaz

**Comment by LaurenceJJones** [comment]
Update as of Ubuntu 24.04 they have the packages back into the main repository 🤷🏻 so I don't know if they are re supporting it or they just back ported Debian packages and it was included. Since there is no "official" post about this we are unsure, since CrowdSec now uses a single package system for all distros based on Debian you will be able to run `apt install crowdsec-nginx-bouncer` and it will tell you at install time if the packages are absent.

**Comment by PrzemekSkw** [comment]
Hello, will be some fix?
```
~# apt install crowdsec-nginx-bouncer
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libnginx-mod-http-lua : Depends: libnginx-mod-http-ndk but it is not going to be installed
                         Depends: nginx-abi-1.24.0-1
E: Unable to correct problems, you have held broken packages.
```
Regards

**Comment by LaurenceJJones** [comment]
> Hello, will be some fix?
> ```
> ~# apt install crowdsec-nginx-bouncer
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> Some packages could not be installed. This may mean that you have
> requested an impossible situation or if you are using the unstable
> distribution that some required packages have not yet been created
> or been moved out of Incoming.
> The following information may help to resolve the situation:
> 
> The following packages have unmet dependencies:
>  libnginx-mod-http-lua : Depends: libnginx-mod-http-ndk but it is not going to be installed
>                          Depends: nginx-abi-1.24.0-1
> E: Unable to correct problems, you have held broken packages.
> ```
> Regards


We can't fix it, the issue is Ubuntu removed lua support in 22 and 23 but then readded it for 24. So the fix is upgrading to 24.04 which is lts.
